models/articles.model.js

```diff
@@ -38,6 +39,11 @@ function selectArticles(sort_by = "created_at", order = "desc") {
   }
 
   return db.query(queryStr).then(({ rows }) => {
+    if (topic) {
+      return rows.filter((row) => {
+        return row.topic === topic;
```

Comment:
we want the SQL string to handle this - you can add a WHERE clause conditionally to it to filter topics from the DB.